### PR TITLE
Cartoon effect 

### PR DESCRIPTION
**Pull request by RajS18**
Here **cartoonize** and **comic** functions implement the cartoon and animated effects on the user provided images resulting in 3 output files (as a result of 1 effect's call). A variety of cartoon effects can be generated for a single input (so as to demonstrate). 
Results are really mesmerizing (by local survey on different images) [added in assets directory].

### CODE REVIEW

File: cartoon_effects.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+args = vars(ap.parse_args())
+# reading the image
+img = cv2.imread((args["image"]))
+
+
+print("Wait, Work is in Progess.")
+
+res_img1,res_img2 = cartoonize(img)
+res_img3 = comic(img)
+cv2.imwrite("rajop/cartoon1.jpg", res_img1)
```
Comment (smaranjitghose) [inline cartoon_effects.py]: As a script intended for general use, I would suggest modifying the output path as 'assets/cartoon_effect.jpg'

File: cartoon_effects.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import cv2 #image processing lib
+import numpy as np #matrix manipulation
+import time 
+
+def comic(img):
+    # do edge detection on a grayscale image
```
Comment (smaranjitghose) [inline cartoon_effects.py]: Nice work with the documentation!

**Review by smaranjitghose — CHANGES_REQUESTED** [review]

**Comment by smaranjitghose** [comment]
Also, try to put the comic cartoon effect in a different script

File: cartoon_effects.py (hunk 2 above)
Comment (RajS18) [reply cartoon_effects.py]: Good to here sir.

File: cartoon_effects.py (hunk 1 above)
Comment (RajS18) [reply cartoon_effects.py]: On it sir.

**Comment by RajS18** [comment]
Greetings Sir,
Please review the changes and if everything is fine merge them, else i'll update accordingly.
Thank you.

File: comic_cartoon_effect.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+#construct the argument parse and parse the arguments
+ap = argparse.ArgumentParser()
+ap.add_argument("-i", "--image", required=True, help="path to input image")
+args = vars(ap.parse_args())
+# reading the image
+img = cv2.imread((args["image"]))
+
+
+print("Wait, Work is in Progess.")
```
Comment (smaranjitghose) [inline comic_cartoon_effect.py:52]: You can replace this with progress bars using tqdm. Try it in another PR, once this gets merged

File: cartoon_effects.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    color1 = cv2.medianBlur(color,25)
+    
+    
+    # 3) Cartoon
+    cartoon = cv2.bitwise_and(color1, color1, mask=edges)
+    return cv2.medianBlur(cartoon,3),edgesOnly
+ 
+          
+# Driver's code 
+#img = cv2.imread("assets/kolkata.jpg")
```
Comment (smaranjitghose) [inline cartoon_effects.py]: Can you please remove this? We need lots of comments but only for explaining our code 😊

Comment (RajS18) [reply cartoon_effects.py]: Sure Sir.

**Comment by RajS18** [comment]
Greetings Sir.
Update the Python script as per requirement.
Please review the code and if everything's fine please do the merge and accept the PR.

Thank you Sir.

**Pull request merged**